polkadot/runtime/parachains/src/paras/mod.rs

```diff
@@ -1149,6 +1164,78 @@ pub mod pallet {
 			MostRecentContext::<T>::insert(&para, context);
 			Ok(())
 		}
+
+		/// Sets the storage for the authorized current code hash of the parachain.
+		///
+		/// This can be useful when we want to trigger `Paras::force_set_current_code(para, code)`
+		/// from a different chain than the one where the `Paras` pallet is deployed.
+		///
+		/// The main reason is to avoid transferring the entire `new_code` wasm blob between chains.
+		/// Instead, we authorize `new_code_hash` with `root`, which can later be applied by
+		/// `Paras::apply_authorized_force_set_current_code(para, new_code)` by anyone.
+		#[pallet::call_index(9)]
+		#[pallet::weight(<T as Config>::WeightInfo::authorize_force_set_current_code_hash())]
+		pub fn authorize_force_set_current_code_hash(
+			origin: OriginFor<T>,
+			para: ParaId,
+			new_code_hash: ValidationCodeHash,
+		) -> DispatchResult {
+			ensure_root(origin)?;
+
+			// if one is already authorized, means it has not been applied yet, so we just replace
+			// it.
+			if let Some(already_authorized) = AuthorizedCodeHash::<T>::take(para) {
+				log::warn!(
+					target: LOG_TARGET,
+					"Already authorized code hash: {:?} found for para {:?}, just removing it!",
+					already_authorized, para
+				);
+			}
+
+			// TODO: FAIL-CI - more validations?
+			// do we need to check against `FutureCodeHash`, `CodeHashRef`,
+			// `PastCodeHash`,... code hashes?
+
+			// insert authorized code hash.
+			AuthorizedCodeHash::<T>::insert(&para, new_code_hash);
+			Self::deposit_event(Event::CodeAuthorized { para_id: para, code_hash: new_code_hash });
+
+			Ok(())
+		}
+
+		/// Applies the already authorized current code for the parachain,
+		/// triggering the same functionality as `force_set_current_code`.
+		#[pallet::call_index(10)]
+		#[pallet::weight(<T as Config>::WeightInfo::apply_authorized_force_set_current_code(new_code.0.len() as u32))]
+		pub fn apply_authorized_force_set_current_code(
+			_origin: OriginFor<T>,
+			para: ParaId,
+			new_code: ValidationCode,
+		) -> DispatchResultWithPostInfo {
+			// no need to ensure, anybody can do this
+
+			// check `new_code` is authorized
+			let Some(authorized_code_hash) = AuthorizedCodeHash::<T>::take(&para) else {
+				log::error!(target: LOG_TARGET, "No authorized code hash found for para {:?}!", para);
+				return Err(Error::<T>::CannotUpgradeCode.into())
```

Comment:
```suggestion
				return Err(Error::<T>::CannotUpgradeCode.into())
```


Just introduce a proper error variant.